app/src/main/java/com/bashkevich/androidfundamentals/movieslist/viewmodel/MoviesListViewModel.kt

```diff
@@ -1,29 +1,30 @@
 package com.bashkevich.androidfundamentals.movieslist.viewmodel
 
 import android.util.Log
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
-import com.bashkevich.androidfundamentals.model.MoviesRepository
+import androidx.lifecycle.*
+import com.bashkevich.androidfundamentals.model.repository.MoviesRepository
 import com.bashkevich.androidfundamentals.model.viewobject.Movie
-import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
 
 class MoviesListViewModel(private val moviesRepository: MoviesRepository) : ViewModel() {
 
-    val moviesListLiveData: LiveData<List<Movie>> = moviesRepository.getAllMovies()
+    var moviesListLiveData: LiveData<List<Movie>>? = null
+    private val _errorLiveData = MutableLiveData<String>()
+
+
+    val errorLiveData: LiveData<String>
+        get() = _errorLiveData
 
     fun loadMoviesList() {
         viewModelScope.launch {
             try {
-                moviesRepository.refreshMovies()
+                moviesListLiveData = moviesRepository.getAllMovies().asLiveData()
```

Comment:
You can subscribe and repost received data to live data that is visible in UI

Reply:
I don't understand what you mean

Reply:
Create kive data and subscribe to it in UI. Here subscribe to data coming from the repository and post it immediately to that live data.